src/System.Data.SqlClient/src/System/Data/Common/AdapterUtil.cs

```diff
@@ -1071,5 +1071,73 @@ static internal Version GetAssemblyVersion()
 
             return s_systemDataVersion;
         }
+
+
+        static internal readonly string[] AzureSqlServerEndpoints = {Res.GetString(Res.AZURESQL_GenericEndpoint),
+                                                                    Res.GetString(Res.AZURESQL_GermanEndpoint),
+                                                                    Res.GetString(Res.AZURESQL_UsGovEndpoint),
+                                                                    Res.GetString(Res.AZURESQL_ChinaEndpoint) };
+
+        static internal bool IsAzureSqlServerEndpoint(string dataSource)
+        {
+            Debug.Assert(!string.IsNullOrWhiteSpace(dataSource));
+
+            string serverName = ExtractServerName(dataSource);
```

Comment:
IsAzureSqlServerEndpoint() method is also executed with NP protocol dataSource, and it returns wrong value because it only tcp  dataSouce correctly. That wrong value is always sent to native SNI when protocol is NP. This is potential problem in the future if someone utilizes isAzureSqlServerEndpoint value for processing NP business logic in native SNI.